Error in LMStudio models parsing

### Describe the bug / provide steps to reproduce it

Summary:
LMStudio model response cannot be parsed when a model in the list missed the `quantization` parameter. 

```
Caused by:
    missing field `quantization` at line 24 column 5
2025-01-28T10:13:16.728349+01:00 [ERROR] Unable to parse LM Studio models response
```

Steps to trigger the problem:
1. Download SmolVLM model in LMStudio
2. Refresh Zed LMStudio with the `connect` button
3. Open zed log 

Actual Behavior:
Zero model are listed from LMStudio because only one is not suitable.
Expected Behavior:
Filter for not compatible models and display the one that match the necessary parameter. 

LMStudio response to the models query : 
```
2025-01-28 10:14:38 [DEBUG] 
Received request: GET to /api/v0/models
2025-01-28 10:14:38  [INFO] 
Returning {
  "data": [
    {
      "id": "deepseek-r1-distill-llama-8b",
      "object": "model",
      "type": "llm",
      "publisher": "lmstudio-community",
      "arch": "llama",
      "compatibility_type": "gguf",
      "quantization": "Q8_0",
      "state": "loaded",
      "max_context_length": 131072,
      "loaded_context_length": 4096
    },
    {
      "id": "smolvlm-500m-instruct",
      "object": "model",
      "type": "vlm",
      "publisher": "mlx-community",
      "arch": "idefics3",
      "compatibility_type": "mlx",
      "state": "not-loaded",
      "max_context_length": 8192
    },
    {
      "id": "text-embedding-nomic-embed-text-v1.5",
      "object": "model",
      "type": "embeddings",
      "publisher": "nomic-ai",
      "arch": "nomic-bert",
      "compatibility_type": "gguf",
      "quantization": "Q4_K_M",
      "state": "not-loaded",
      "max_context_length": 2048
    }
  ],
  "object": "list"
}
```


### Zed Version and System Specs

Zed: v0.170.4 (Zed)
OS: macOS 15.2.0
Memory: 128 GiB
Architecture: aarch64

--- a/crates/lmstudio/src/lmstudio.rs
+++ b/crates/lmstudio/src/lmstudio.rs
@@ -192,7 +192,7 @@ pub struct ModelEntry {
     pub publisher: String,
     pub arch: Option<String>,
     pub compatibility_type: CompatibilityType,
-    pub quantization: String,
+    pub quantization: Option<String>,
     pub state: ModelState,
     pub max_context_length: Option<u32>,
     pub loaded_context_length: Option<u32>,

